### ISSUE TITLE
Memory leak in the implementation of `RequestWithCache`.

### ISSUE THREAD

**Issue by happy-san** [typesense/typesense-js#66]
## Description

In the [current implementation](https://github.com/typesense/typesense-js/blob/master/src/Typesense/RequestWithCache.js#L16), a cache entry is only deleted when it is requested and it has expired. If the cache entry is never requested, it would never be deleted. Combined with the fact `_responseCache` has no limit on how many entries it can store, we've got a leak.

```javascript
    if (cacheEntry) {
      if (Date.now() - cacheEntry.requestTimestamp < cacheResponseForSeconds * 1000) {
        // Cache entry is still valid, return it
        return Promise.resolve(cacheEntry.response)
      } else {
        // Cache entry has expired, so delete it explicitly
        delete this._responseCache[requestFunctionArgumentsJSON]
      }
    }
```

## Steps to reproduce

1. [Enable caching of results](https://github.com/typesense/typesense-js/blob/master/src/Typesense/Configuration.js#L24)
2. Keep searching for stuff without ending the session, memory usage would steadily increase.

## Expected Behavior

There should be an eviction policy in place to get rid of the unwanted cached entries and preferably an upper limit on how many results can be cached. I'd recommend implementing Least Recently Used algorithm.

**Comment by jasonbosco** [comment]
@dcantu476 Yes it is. Mainly a concern if users search say several 100 times without a page refresh in a single session.

Could happen more often in a React Native environment for example.

Definitely want to address it. Do you want take a stab at it?
